### PR TITLE
Option to position and size components with two corners.

### PR DESCRIPTION
**Pull request by mtogstad**
This lets you lock a component to a specific bounding rectangle, which allows for precise tiling during zoom with the experimental google maps API.

I work on a project that uses a technique similar to the thousands-of-markers sample project linked to from the readme.  If you run that project against the 3.32 google maps API you can see the same issue I was having.  The tiles are not scaled when frames are drawn during a zoom, which causes the edges to either have gaps between them, or to overlap.  This adds the option to specify a second LatLng (the southeast corner) to give a component a bounds-locked box to render into.

Here is a before example:
giphy

And after:
giphy-1

### CODE REVIEW

File: src/google_map_markers.js [hunk 1]
```diff
@@ -269,11 +269,30 @@ export default class GoogleMapMarkers extends Component {
...
-        const stylePtPos = {
+        let stylePtPos = {
           left: pt.x,
           top: pt.y,
         };
 
+        if (
+          child.props.seLatLng !== undefined ||
+          (
+            child.props.seLat !== undefined &&
```
Comment (itsmichaeldiego) [inline src/google_map_markers.js]: @mtogstad Could you explain briefly what are you doing here and why? Maybe leave a comment

**Comment by itsmichaeldiego** [comment]
@stephenfarrar What do you think about this? Given its part of the code you've been working with.

**Comment by stephenfarrar** [comment]
Looks good to me.

**Comment by mtogstad** [comment]
Just pushed an explanation, and thanks for looking at this pull request so fast!

**Comment by itsmichaeldiego** [comment]
@mtogstad Thank you very much for this! Please update to `1.0.3` and let me know how it works. Feel free to make more PRS, we need more help!

**Comment by palashkaria** [comment]
@mtogstad Hey, I know that this issue is closed, but I'm curious, how did this help with the canvas markers? I am using it for a similar use case & facing the exact same problem. Could you share the relevant change you made in this repo, which fixed the issue? https://github.com/istarkov/google-map-thousands-markers

**Comment by palashkaria** [comment]
I'm on 1.0.5 itself, but the issue is not with this lib, it's with the canvas implementation (https://github.com/istarkov/google-map-thousands-markers). 

What I understood from this change is that I have to pass `seLatLng` to `CanvasTile` in that implementation. But that doesn't seem to work..

**Comment by itsmichaeldiego** [comment]
@palashkaria Oh yes, what changed is that you can pass either `seLatLng={[lat, lng]}` or `seLat={lat} seLng={lng}`

**Comment by palashkaria** [comment]
I think it's like `seLatLng={{lat, lng}}`. But question is, specifically how does it help in those before/after images posted above, from the 'google-map-thousands-markers' example. I'm trying something here, but it doesn't seem to work..

https://github.com/palashkaria/google-map-thousands-markers/tree/google-maps-v3.32-fix

**Comment by itsmichaeldiego** [comment]
@palashkaria What are you trying? Could you provide a live example?

**Comment by itsmichaeldiego** [comment]
@palashkaria That should work given @mtogstad's change, but there might be a chance that you dont need that `seLatLng` anymore as we went through some interesting changes in the way markers are drawn. What exactly you want to accomplish? Could you provide a live example?

I think the best person to answer would be @mtogstad, as he did the change and tested it locally.

**Comment by punnone** [comment]
@palashkaria I'm facing the same problem. Did you get a solution?

**Comment by Heedster** [comment]
Fixed it. 2 things have to be done.
1. Adding a seLatLng to the tile [CanvasMap.js]
```
tiles[key] = {
        ...tile2LatLng({ x, y }, zoom),
        seLatLng: { ...tile2LatLng({ x: x + 1, y: y + 1 }, zoom) },
        x,
        y,
        markers: []
   };
```
2. Remove explicit width from the canvas tile [CanvasTile.js]
```
const canvasStyle = {
  width: "100%",
  height: "100%"
};
```

Updated the codesanbox for reference: https://codesandbox.io/s/6v5vx7z65w

**Pull request merged**